secure_message/resources/messages.py

```diff
@@ -45,15 +45,26 @@ def post(self):
         post_data['from_internal'] = g.user.is_internal
         message = self._validate_post_data(post_data)
 
-        if message.errors == {}:
-            logger.info("Message passed validation")
-            self._message_save(message)
-            # listener errors are logged but still a 201 reported
-            MessageSend._alert_listeners(message.data)
-            return make_response(jsonify({'status': '201', 'msg_id': message.data.msg_id, 'thread_id': message.data.thread_id}), 201)
-
-        logger.error('Message send failed', errors=message.errors)
-        return make_response(jsonify(message.errors), 400)
+        if message.errors != {}:
+            logger.error('Message send failed', errors=message.errors)
+            return make_response(jsonify(message.errors), 400)
```

Comment:
Is this definitely due to a bad request and not a server error? Would suggest 500 is more valid here (if not being more specific with our error handling than `{}`!!). There's also an argument that the Invalid claim should be 404 or 422 but it seems like there's been a move away from 400 indicating bad syntax of late :)

Reply:
I think 400 seems about right.  Even if it is wrong, the code hasn't changed but moved as the logic is a bit different.  I'd leave it as is, as something might be relying on it being 400 specifically and we can raise a card to sort it if it turns out that it's wrong.